### DISCUSSION TITLE
Input format of the model

### DISCUSSION THREAD

**Discussion by Giustino98** [dscripka/openWakeWord#28] [Q&A] [answered]
Hi everyone,

I am trying to use pre-trained model to make inference on audio record acquired from vue.js front-end. My question is: is there any documentation about how the input tensor needs to be?
I tried to acquire microphone record with tensorflow.js and pass its spectrogram to python backend but it looks very different than the audio directly acquired in python examples provided so the output of the model is always 0.
My ideal result would be the hugging face demo: https://huggingface.co/spaces/davidscripka/openWakeWord

Any help would be very appreciated!

**Comment by dscripka** [comment, marked as answer]
Hi @Giustino98 , this is a good question, and I agree that the current documentation doesn't fully describe how the data passes through the different models in openWakeWord. At a high level, this is the process:

1) Raw audio data (16-bit PCM format, or in python an array/list of 16-bit integer data) is passed to a model that computes the melspectrogram. openWakeWord then does a [simple linear transformation](https://github.com/dscripka/openWakeWord/blob/3ced4cf59ff04692a337dfed992b9b3559d6c928/openwakeword/utils.py#L100) of the melspectrogram.

2) The melspectogram is then passed to the audio embedding model. This produces an array of shape (1, time, 96), where time is the number of frames that the particular wakeword model is trained to use.

3) The audio embeddings are then passed to the classification models for the wakewords. These produce the final scores between 0-1.

From your comment, it sounds like you want to collect the audio data via javascript and then pass this to openWakeWord. There is nothing wrong with this approach, you just need to ensure that the audio data is converted to 16-bit integer arrays before passing it to the openWakeWord library.
